Changes to support easier deployment while maintaining a quick dev cycle

pontus opened the pull request.

pontus committed:
Placeholder files for config directory

pontus committed:
No special development treatment

pontus committed:
New files for deployment

pontus committed:
Container build changes for deployment

pontus committed:
Changes to ease deployment

pontus committed:
Make sure development environment works more consistently

pontus committed:
Compute backend URL correctly)

pontus committed:
Run registration action with the setup in container instead

pontus committed:
Disable secure cookies for now

norling commented on frontend/communication.tsx:
```diff
@@ -2,15 +2,9 @@ import React from 'react'
-  process.env.NODE_ENV === 'development'
-  ? 'http://localhost:4200'
-  : '';
+export const backend_url = window.location.href.slice(0, -window.location.pathname.length);
```
behöver man sätta backend_url alls här? om vi bara kör samma så går det ju automagiskt till rätt host? eller finns det nån säkerhetsanledning att ha absoluta url'er? isåfall kan du använda `location.origin` direkt för att få url'en iaf!

pontus replied:
`location.origin` är nog bättre, ja. Men nä, det borde nog inte behövas och las nog mest in för att det fanns.

Jag ser ingen anledning att man skulle behöva ha den absolut.

norling committed:
Add config location via environment variable

norling committed:
remove special case backend url

pontus committed:
Use bash for entryscript to allow {a,b,c} usage

pontus committed:
Fix command for parcel server

pontus committed:
Copy configuration (if any) to temporary storage and use that

Currently, the WSGI code will try to provide a configuration if
none is provided. If we point the configuration environment variable
(HERDBOOK_CONFIG) to the /config directory, the server may not be able
to create it there for permissions reasons (or r/o).

Copy any configuration to /tmp and use that instead.

pontus committed:
Make sure wsgi server can read configuration

jonandernovella commented:
Looks good. Perhaps all the "fixup changes" should be squashed.
